metric_providers/psu/energy/ac/xgboost/machine/provider.py

```diff
@@ -36,6 +37,15 @@ def get_stderr(self):
     def start_profiling(self, containers=None):
         self._has_started = True
 
+    #TODO: not a fan of using the full key name here. any way to avoid this?
+    # keeping original checks in read_metrics for now
+    def check_system(self):
+        config = GlobalConfig().config
+        if 'cpu.utilization.procfs.system.provider.CpuUtilizationProcfsSystemProvider' not in config['measurement']['metric-providers']['linux']:
+            raise MetricProviderConfigurationError(f"{self._metric_name} provider could not be started.\nPlease activate the CpuUtilizationProcfsSystemProvider in the config.yml\n \
+                This is required to run PsuEnergyAcSdiaMachineProvider")
+        self.check_parallel_provider_running()
```

Comment:
I do see that you cannot call the base super function here. But what "parallel provider" are you checking here for?

There is no binary or script to execute. Actually this is even a logical bug in the code already. Neither the base check_system nor the check_parallel_provider_running may be called. Also a comment should clarify that we do that on purpose